letshelp-letsencrypt/letshelp_letsencrypt/apache.py

```diff
@@ -0,0 +1,303 @@
+#!/usr/bin/env python
+"""Let's Encrypt Apache configuration submission script"""
+import argparse
+import atexit
+import contextlib
+import os
+import re
+import shutil
+import subprocess
+import sys
+import tarfile
+import tempfile
+import textwrap
+
+
+_DESCRIPTION = """
+Let's Help is a simple script you can run to help out the Let's Encrypt
+project. Since Let's Encrypt will support automatically configuring HTTPS on
+many servers, we want to test this functionality on as many configurations as
+possible. This script will create a sanitized copy of your Apache
+configuration, notifying you of the files that have been selected. If (and only
+if) you approve this selection, these files will be sent to the Let's Encrypt
+developers.
+
+"""
+
+
+_NO_APACHECTL = """
+Unable to find `apachectl` which is required for this script to work. If it is
+installed, please run this script again with the --apache-ctl command line
+argument and the path to the binary.
+
+"""
+
+
+# Keywords likely to be found in filenames of sensitive files
+_SENSITIVE_FILENAME_REGEX = re.compile(r"^(?!.*proxy_fdpass).*pass.*$|private|"
+                                       r"secret|cert|crt|key|\.pem|\.der|rsa|"
```

Comment:
These probably shouldn't be in the Apache configuration, but just in case...

\.p12|\.pfx|\..p7b